### ISSUE TITLE
Error parsing when declaring a module using shorthand ambient modules

### ISSUE THREAD

**Issue by ablamunits** [typescript-eslint/typescript-eslint#1726]
**What code were you trying to parse?**

I have a file named `global.d.ts` in which I would like to use [shorthand ambient modules](https://www.typescriptlang.org/docs/handbook/modules.html#shorthand-ambient-modules) in the following way:

```ts
declare module '*';
```

**What did you expect to happen?**

I expect no error.

**What actually happened?**

I get the following error:

```
  6:19 error  Parsing error: Unexpected token, expected "{"
```

The unexpected token is pointing at `;` as the problem.

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `2.20.0` |
| `TypeScript`                | `3.7.5` |
| `ESLint`                    | `5.12.0` |
| `node`                      | `12.10.0` |
| `npm`                       | `6.10.3` |

**Comment by bradzacher** [comment]
I'm unable to reproduce this.
Your example parses fine against master.
Could you please provide an small example repo where it reproduces?

**Comment by ablamunits** [comment]
Hi @bradzacher - thanks for the quick reply.
It turns out that this was a misconfiguration on my part 🤦‍♂ 
We are currently in the process of migrating some packages in a monorepo to TS and in this case the eslint config was set to use `babel-eslint` parser instead of `@typescript-eslint/parser`.

Thanks for taking the time to look into it, I appreciate it.
